src/types/complex.rs

```diff
@@ -155,12 +146,13 @@ mod not_limited_impls {
     impl<'py> Neg for Borrowed<'_, 'py, PyComplex> {
         type Output = Bound<'py, PyComplex>;
         fn neg(self) -> Self::Output {
-            unsafe {
-                let val = (*self.as_ptr().cast::<ffi::PyComplexObject>()).cval;
-                ffi::PyComplex_FromCComplex(ffi::_Py_c_neg(val))
-                    .assume_owned(self.py())
-                    .downcast_into_unchecked()
-            }
+            self.as_borrowed()
+                .getattr("__neg__")
+                .expect("Cannot find `__neg__` method.")
+                .call0()
+                .expect("Failed to call `__neg__` method.")
+                .downcast_into()
+                .expect("Failed to downcast to complex number.")
```

Comment:
I think there are subtleties around calling the dunder method and so it's probably better to go through `PyNumber_Negative` ffi call.

(Open question if we should add `PyAnyMethods::neg` and `PyAnyMethods::pos` to expose that functionality to everyone.)

Reply:
I think adding those methods to the public will be useful. I can have a look at them tomorrow